File: src/ex002.rs
fn fib(n: i32) -> i32 {
    let mut a = 1; let mut b = 1;
    for _ in 0..(n-1) {
        let c = a;
        a = b;
        b = b + c;
    }
    a
}

fn main() {
    let mut s = 0;
    let mut n = 1;

    loop {
        if n % 3 == 0 {     // f % 2 == 0 <=> n % 3 == 0
            println!("{}", n);
            let f = fib(n);
            if f > 4000000 {
                break;
            }
            s += f;
        }
        n += 1;
    }
    println!("{}", s);
}

File: src/ex002.scala
object Ex002 {
  def fib(n: Int): Int =
    if (n <= 1) n else fib(n - 1) + fib(n - 2)
  def compute(maxNum: Int, maxIndex: Int): Int =
    (0 to maxIndex).map(fib).filter(f => f % 2 == 0).takeWhile(f => f <= maxNum).sum
  def main(args: Array[String]) {
    val n = 40
    println(fib(n) > 4000000) // => true
    println(compute(4000000, n)) // => 4613732
  }
}

